File: backend/src/main/kotlin/com/example/backend2/global/messaging/subscriber/RedisMessageSubscriber.java
package com.example.backend2.global.messaging.subscriber;

import lombok.RequiredArgsConstructor;
import org.example.bidflow.global.messaging.dto.MessagePayload;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
public class RedisMessageSubscriber implements MessageSubscriber<MessagePayload> {

    @Override
    public void subscribe(String topic) {

    }

    @Override
    public void onMessage(String topic, MessagePayload payload) {
        System.out.println("Redis 구독 메시지 수신: " + payload);
    }
}

File: backend/src/main/kotlin/com/example/backend2/global/messaging/subscriber/RedisMessageSubscriber.kt
package com.example.backend2.global.messaging.subscriber

import com.example.backend2.global.messaging.dto.MessagePayload
import org.springframework.stereotype.Component

@Component
class RedisMessageSubscriber : MessageSubscriber<MessagePayload> {
    override fun subscribe(topic: String) {
    }

    override fun onMessage(
        topic: String,
        payload: MessagePayload,
    ) {
        println("Redis 구독 메시지 수신: $payload")
    }
}
